### PR TITLE
adding more docs as pre migration of gudocs and scripts to run it locally

### PR DESCRIPTION
**Pull request by mkuzdowicz**
adding more docs as pre migration of gudocs to ed tools and making running app locally easier

- adding script to run the app locally

adding more docs about:

- what REDIS is used for here
- how to run app locally connect it to any google account and be able to perform some manual tests

### CODE REVIEW

File: docs/redisUsage.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
...oasis.opendocument.spreadsheet\":\"https://d...\"},\"userPermission\":{\"kind\":\"drive#permission\",\"etag\":\"\\\"43675648\\\"\",\"id\":\"me\",\"selfLink\":\"https://www.googleapis.com/drive/v2/files/11111/permissions/me\",\"role\":\"writer\",\"type\":\"user\"},\"quotaBytesUsed\":\"0\",\"ownerNames\":[\"John Doe\"],\"owners\":[{\"kind\":\"drive#user\",\"displayName\":\"John Doe\",\"picture\":{\"url\":\"https://lh3.googleusercontent.com/a-/...\"},\"isAuthenticatedUser\":false,\"permissionId\":\"....\",\"emailAddress\":\"j.d@guardian.co.uk\"}],\"lastModifyingUserName\":\"John Doe\",\"lastModifyingUser\":{\"kind\":\"drive#user\",\"displayName\":\"John Doe\",\"picture\":{\"url\":\"https://lh3.googleusercontent.com/a-/...\"},\"isAuthenticatedUser\":false,\"permissionId\":\"....\",\"emailAddress\":\"j.d@guardian.co.uk\"},\"capabilities\":{\"canCopy\":true,\"canEdit\":true},\"editable\":true,\"copyable\":true,\"writersCanShare\":true,\"shared\":true,\"explicitlyTrashed\":false,\"appDataContents\":false,\"spaces\":[\"drive\"]},\"lastUploadTest\":\"2020-01-15T09:45:24.506Z\",\"lastUploadProd\":\"2020-01-15T07:22:25.257Z\",\"domainPermissions\":\"none\",\"properties\":{\"isTable\":false}}"
```
Comment (AWare) [inline docs/redisUsage.md]: Might want to get rid of the escapes and use a ` ``` ` section

**Comment by AWare** [comment]
I think these changes should be fine for master, but now's a good time to start our migration branch. 👍

File: docs/redisUsage.md (hunk 1 above)
Comment (mkuzdowicz) [reply docs/redisUsage.md]: done

File: docs/redisUsage.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+Redis is used here as a main persitence source
+======================
+
+It persists the data that we can see on the https://visuals.gutools.co.uk/docs/ page
+
+so mainly the `gudocs:index` key have the keys of the files that we see on the table on web page, `gudocs` have the details about last modified
+
+once you log into the machine that runs redis youc an use redis-cli to perform queries
```
Comment (akash1810) [inline docs/redisUsage.md]:
```suggestion
once you log into the machine that runs redis you can use redis-cli to perform queries
```

File: docs/redisUsage.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+`redis-cli ZRANGE gudocs:index 0 -1 WITHSCORES` that will list the files ids, example response
+
+    1) "111111"
+    2) "222222"
+
+and then you can query the file details by `redis-cli GET gudocs:111111`, example reposne:
```
Comment (akash1810) [inline docs/redisUsage.md]:
```suggestion
and then you can query the file details by `redis-cli GET gudocs:111111`, example response:
```

File: docs/runningLocally.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+## Requirements:
+1. Google account with google drive
+2. "Service account" key for google API project
+3. Janus creds for composer account so that you will have access to s3 bucket `gudocs-dev`
+
+## connecting to Google account and google app project
+
+- you can connect app to any google account for testing purpose 
+- the app is using google drive changes API https://developers.google.com/drive/api/v2/manage-changes to track the changes in the fiels
```
Comment (akash1810) [inline docs/runningLocally.md]:
```suggestion
- the app is using google drive changes API https://developers.google.com/drive/api/v2/manage-changes to track the changes in the files
```

File: docs/runningLocally.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+- getting acces to google drive API
```
Comment (akash1810) [inline docs/runningLocally.md]:
```suggestion
- getting access to google drive API
```

File: docs/runningLocally.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+look at instructions in [googleApis.md](/docs/googleApis.md) file to see instructions on how to do that
+
+## how to track google documents by the app
+
+When you create a "Service account" key an email address for that project will be generated
+in a form `<key-name>@<project-name>.iam.gserviceaccount.com`
+
+then once you create any doc in google docs or sheets you will need to share it with the email address of the google api project `<key-name>@<project-name>.iam.gserviceaccount.com`
+
+after that the app should track all the changes/editis in any file you share
```
Comment (akash1810) [inline docs/runningLocally.md]:
```suggestion
after that the app should track all the changes/edits in any file you share
```

File: docs/runningLocally.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+## preforming simple manual test
+
+you will need redis running locally 
+
+run `./dev-start-ui.sh` and you will see the UI at http://localhost:4001/ that lists the files (it should be blank once you run it for the first time with a freash google account)
```
Comment (akash1810) [inline docs/runningLocally.md]:
```suggestion
run `./dev-start-ui.sh` and you will see the UI at http://localhost:4001/ that lists the files (it should be blank once you run it for the first time with a fresh google account)
```

File: docs/runningLocally.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+run `./dev-start-fetch.sh` that will hit google drive changes api and update locally running REDIS with information about google docs file changes
+
+then you should see details abput the files that you shared with the app email `<key-name>@<project-name>.iam.gserviceaccount.com`
```
Comment (akash1810) [inline docs/runningLocally.md]:
```suggestion
then you should see details about the files that you shared with the app email `<key-name>@<project-name>.iam.gserviceaccount.com`
```

File: docs/runningLocally.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+editi something in the file in google docs/sheets shared with `<key-name>@<project-name>.iam.gserviceaccount.com`
```
Comment (akash1810) [inline docs/runningLocally.md]:
```suggestion
edit something in the file in google docs/sheets shared with `<key-name>@<project-name>.iam.gserviceaccount.com` 
```

File: docs/runningLocally.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+you will need redis running locally 
+
+run `./dev-start-ui.sh` and you will see the UI at http://localhost:4001/ that lists the files (it should be blank once you run it for the first time with a freash google account)
+
+run `./dev-start-fetch.sh` that will hit google drive changes api and update locally running REDIS with information about google docs file changes
+
+then you should see details abput the files that you shared with the app email `<key-name>@<project-name>.iam.gserviceaccount.com`
+
+editi something in the file in google docs/sheets shared with `<key-name>@<project-name>.iam.gserviceaccount.com` 
+un `./dev-start-fetch.sh` again
```
Comment (akash1810) [inline docs/runningLocally.md]:
```suggestion
run `./dev-start-fetch.sh` again 
```

File: src/fetch.js [hunk 11]
```diff
@@ -2,14 +2,20 @@ import gu from 'koa-gu'
 import co from 'co'
 import fileManager from './fileManager'
 import program from 'commander'
+import AWS from 'aws-sdk'
+
+// setup aws credentials
+gu.init({www:false});
+var credentials = new AWS.SharedIniFileCredentials(gu.config.aws_profile);
```
Comment (akash1810) [inline src/fetch.js]: can we use `const`?

```suggestion
const credentials = new AWS.SharedIniFileCredentials(gu.config.aws_profile);
```

File: dev-start-fetch.sh [hunk 12]
```diff
@@ -0,0 +1,8 @@
+export GU_aws_profile="composer"
```
[the remark this answers is not in the document]
Comment (AWare) [reply dev-start-fetch.sh]: These env vars are read by github.com/guardian/koa-gu/ which we don't want to make too many changes to.

File: src/fetch.js (hunk 11 above)
Comment (mkuzdowicz) [reply src/fetch.js]: we do not have CI/CD now to deply apps on visuals ec2 so will keep it as was before, that is mainly code move

File: dev-start-fetch.sh (hunk 12 above)
[the remark this answers is not in the document]
Comment (mkuzdowicz) [reply dev-start-fetch.sh]: yes that will be great to do later, we will do some refactor anyway while migrating the app to lambda later

[the remark this answers is not in the document]
Comment (mkuzdowicz) [reply dev-start-fetch.sh]: yeah we can think about it later

**Pull request merged**